File: KTAX_SOS_Workflow.WebApp/app/models/api-models/OrdemServico.ts
/// <reference path="../../configs/_all.ts" />
module Consermaq {
    export class OrdemServico {    
        public id: number;
        public servicoSolicitado: string;
        public dataInicial: Date;
        public dataEncerramento: Date;
        public status: any;
        public valorTotal: number;
        public clienteId: number;
        public cliente: Cliente;
        
        public servicos: Array<Servico>;
        public materiaisAtualizados: Array<Produto>;
        public servicoItemsRemovidos: Array<ServicoItem>;
        public servicosRemovidos: Array<Servico>;

        constructor() {       
        }
    }
}

File: ModernWebStore/ModernWebStore.Domain/Entities/OrdemServico.cs
using ModernWebStore.Domain.Enums;
using System;
using System.Collections.Generic;

namespace ModernWebStore.Domain.Entities
{
    public class OrdemServico
    {
        public OrdemServico()
        {
            this.Status = EOrderStatus.Created;
            this.DataInicial = DateTime.Now;
        }

        public int Id { get; set; }

        public string servicoSolicitado { get; set; }
        public DateTime? DataInicial { get; set; }
        public DateTime? DataEncerramento { get; set; }

        public EOrderStatus Status { get; set; }

        public decimal ValorTotal { get; set; }

        public int ClienteId { get; set; }
        public Cliente Cliente { get; set; }

        public ICollection<Servico> Servicos { get; set; }
       
        public ICollection<Product> MateriaisAtualizados { get; set; }
        public ICollection<ServicoItem> ServicoItemsRemovidos { get; set; }
        public ICollection<Servico> ServicosRemovidos { get; set; }


        public void Aberto()
        {
            this.Status = EOrderStatus.Created;
        }

        public void Pendente()
        {
            this.Status = EOrderStatus.Pendente;
        }

        public void Finish()
        {

            this.Status = EOrderStatus.Finish;
        }

        public void Cancel()
        {          
            this.Status = EOrderStatus.Canceled;
        }
    }
}
